### PR TITLE
Move to qgridnext

### PR DESCRIPTION
**Pull request by sarthak-dv**
### :pencil: Description

**Type:** :roller_coaster: `infrastructure`

To avoid any future errors that may be caused by the deprecated `qgrid` library we are temporarily moving to [`qgridnext` ](https://github.com/zhihanyue/qgridnext)until we find a reliable alternative.

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Review by jamesgillanders — APPROVED** [review]
Looks good to me!

**Comment by jamesgillanders** [comment]
@andrewfullard We think it makes sense that this benchmark test fails but we need a second opinion. If it's a problem let us know -- if it isn't, then we will merge once Jaladh has reviewed it

**Comment by andrewfullard** [comment]
Seems that qgridnext doesn't exist in conda, should be a pip install instead

**Comment by sarthak-dv** [comment]
> Seems that qgridnext doesn't exist in conda, should be a pip install instead

Yes, that seems to be the issue here, I've added `pip install qgridnext` in tests.yml, do I need to add it somewhere else for the benchmarks to pass?

File: tardis_env3.yml
Comment (jaladh-singhal) [inline tardis_env3.yml:40]: This won't work because qgridnext is not installable through conda. I got an error when updating/creating tardis env. Explicitly define to use pip:

```suggestion
  - pip:
    - qgridnext
```

File: .github/workflows/tests.yml
Comment (jaladh-singhal) [inline .github/workflows/tests.yml:87]: This shouldn't be needed after the change I suggested in env file. Try removing it

**Review by jaladh-singhal — CHANGES_REQUESTED** [review]
Thank you for this PR @sarthak-dv - I tested the widgets notebooks locally, they work now!

Although, there's a problem with dependency specification - see my comment (it's a small change)

**Comment by jaladh-singhal** [comment]
@sarthak-dv benchmarks "should" pass after the suggested change.

Unrelated, @andrewfullard do we need to update conda lock files as well?

**Comment by andrewfullard** [comment]
> @sarthak-dv benchmarks "should" pass after the suggested change.
> 
> Unrelated, @andrewfullard do we need to update conda lock files as well?

Yes, we will need to regenerate the environment.

File: .github/workflows/tests.yml
Comment (sarthak-dv) [reply .github/workflows/tests.yml:87]: I tried removing this but the tests failed because qgridnext is not installed, this code first installs qgridnext for tests to run. If we don't need this the other option would be to enable `pip install -e . --user` line in the file but that will install all the dependencies again.

**Comment by sarthak-dv** [comment]
The tests in `test_spectrum_solver.py` are showing error, I don’t think it is related to the changes made in this PR. Other than that the benchmarks build is passing after making the changes @jaladh-singhal suggested.

**Comment by andrewfullard** [comment]
> The tests in `test_spectrum_solver.py` are showing error, I don’t think it is related to the changes made in this PR. Other than that the benchmarks build is passing after making the changes @jaladh-singhal suggested.

Yeah that's fine- the test failures are because of changes to regression data in support of another PR

**Comment by andrewfullard** [comment]
Looks like the docs build from the lock file, so they won't work until that is updated too.

File: .github/workflows/tests.yml
Comment (jaladh-singhal) [reply .github/workflows/tests.yml:87]: ok

**Review by jaladh-singhal — APPROVED** [review]
Thanks for the updates. @andrewfullard if you're happy, please merge it.

**Review by jamesgillanders — APPROVED** [review]
Looks good to me -- thanks @sarthak-dv ! I'll leave Andrew to verify the various tests failing are okay, and if so, he can merge.

**Pull request merged**
